src/TesApi.Web/BatchScheduler.cs

```diff
@@ -834,14 +981,21 @@ private async Task<CloudTask> ConvertTesTaskToBatchTaskAsync(TesTask task, bool
             var batchScriptSasUrl = await this.storageAccessProvider.MapLocalPathToSasUrlAsync($"/{batchScriptPath}");
             var batchExecutionDirectorySasUrl = await this.storageAccessProvider.MapLocalPathToSasUrlAsync($"/{batchExecutionDirectoryPath}", getContainerSas: true);
 
-            var cloudTask = new CloudTask(taskId, $"/bin/sh {batchScriptPath}")
+            //var batchRunScriptPath = "task-run.sh";
+            var batchRunScriptContent = string.Join(") && (", File.ReadAllLines(Path.Combine(AppContext.BaseDirectory, "scripts/task-run.sh")))
```

Comment:
ReadAllLinesAsync and store it in a static string, so that we only hit the file system once

Reply:
✅